src/main/java/fr/techad/edc/client/internal/io/HttpReaderImpl.java

```diff
@@ -279,35 +277,40 @@ private DocumentationItemType getType(JsonObject jsonObject) {
 
     }
 
-    private Map<String, String> readLabelsForLang(String languageCode) throws IOException, InvalidUrlException {
-        Map<String, String> labels = Maps.newHashMap();
-        String label;
-
+    private String getLabelUrl(String languageCode) throws InvalidUrlException {
         String relativePath = POPOVER_I18N_PATH + languageCode + I18N_FILE_EXTENSION;
         String labelUrl = StringUtils.appendIfMissing(this.clientConfiguration.getDocumentationUrl(), "/") + relativePath;
         LOGGER.debug("Reading labels for lang {}, url {}, labelUrl {}", languageCode, relativePath, labelUrl);
+        return labelUrl;
+    }
+
+    private I18NContent readi18NContent(String languageCode, I18NContent i18nContent) throws IOException, InvalidUrlException {
+        String label;
+        String labelUrl = getLabelUrl(languageCode);
 
         try {
             // Get the file content (.json) for the language
             label = client.get(labelUrl);
-            LOGGER.debug("Retrieved label: {}", label);
             // Decode Json
             JsonElement jsonContent = parseString(label);
-            if (jsonContent.isJsonObject() && jsonContent.getAsJsonObject().get(I18N_LABELS_ROOT) != null
-                    && jsonContent.getAsJsonObject().get(I18N_LABELS_ROOT).getAsJsonObject().isJsonObject()) {
-                JsonObject jsonObject = jsonContent.getAsJsonObject().get(I18N_LABELS_ROOT).getAsJsonObject();
-                labels = jsonObject.entrySet().stream()
+
+            Set<String> i18NKeys = jsonContent.getAsJsonObject().keySet();
+            JsonObject jsonObjectType = null;
+
+            for (String key : i18NKeys) {
+                jsonObjectType = jsonContent.getAsJsonObject().get(key).getAsJsonObject();
+               labels = jsonObjectType.entrySet().stream()
                         .collect(Collectors.toMap(Map.Entry::getKey, e -> {
                             LOGGER.debug("Creating map entry for key: {} and value {}", e.getKey(), e.getValue().getAsString());
+                            i18nContent.setMessage(languageCode, key, e.getKey(), e.getValue().getAsString());
+
                             return e.getValue().getAsString();
                         }));
             }
         } catch (Error4xxException e) {
-            LOGGER.error("Could not read the labels for the lang {}, err {}", languageCode, e);
+            LOGGER.error("Could not read the label for the lang {}, err {}", languageCode, e);
         }
 
-        LOGGER.debug("Returning labels for lang {}, labels {}", languageCode, labels);
-        return labels;
+        return i18nContent;
```

Comment:
No need to return this content because it is defined as parameter and this method adds translation in it.